server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -0,0 +1,314 @@
+      boolean error = false;
+      for (SortedMap<TServerInstance,TabletServerStatus> s1 : newPools.values()) {
+        for (SortedMap<TServerInstance,TabletServerStatus> s2 : newPools.values()) {
+          if (s1 == s2) {
+            continue;
+          }
+          if (!Collections.disjoint(s1.keySet(), s2.keySet())) {
+            LOG.error("Pools are not disjoint: {}, there is a problem with your regular expressions. Putting all servers in default pool", newPools);
+            error = true;
+          }
+        }
+      }
+      if (error) {
+        // Put all servers into the default pool
+        newPools.clear();
+        newPools.put(DEFAULT_POOL, new TreeMap<TServerInstance,TabletServerStatus>());
+        for (Entry<TServerInstance,TabletServerStatus> e : current.entrySet()) {
+          newPools.get(DEFAULT_POOL).put(e.getKey(), e.getValue());
+        }
+      }
+      pools = newPools;
+      this.lastPoolRecheck = System.currentTimeMillis();
+    }
+    return pools;
+  }
+
+  /**
+   * Matches host against the regexes and returns the matching pool name
+   *
+   * @param host
+   *          tablet server host
+   * @return name of pool. will return default pool if host matches more than one regex
+   */
+  protected String getPoolNameForHost(String host) {
+    String test = host;
+    String table = DEFAULT_POOL;
+    if (!isIpBasedRegex) {
+      try {
+        test = getNameFromIp(host);
+      } catch (UnknownHostException e1) {
+        LOG.error("Unable to determine host name for IP: " + host + ", setting to default pool", e1);
+        return table;
+      }
+    }
+    for (Entry<String,Pattern> e : poolNameToRegexPattern.entrySet()) {
+      if (e.getValue().matcher(test).matches()) {
+        if (!table.equals(DEFAULT_POOL)) {
+          LOG.warn("host {} matches more than one regex, assigning to default pool", host);
+          return DEFAULT_POOL;
+        }
+        table = e.getKey();
+      }
+    }
+    return table;
+  }
+
+  protected String getNameFromIp(String hostIp) throws UnknownHostException {
+    return InetAddress.getByName(hostIp).getHostName();
+  }
+
+  /**
+   * Matches table name against pool names, returns matching pool name or DEFAULT_POOL.
+   *
+   * @param tableName
+   *          name of table
+   * @return tablet server pool name (table name or DEFAULT_POOL)
+   */
+  protected String getPoolNameForTable(String tableName) {
+    if (null == tableName) {
+      return DEFAULT_POOL;
+    }
+    return poolNameToRegexPattern.containsKey(tableName) ? tableName : DEFAULT_POOL;
+  }
+
+  /**
+   * Parse configuration and extract properties
+   *
+   * @param conf
+   *          server configuration
+   */
+  protected void parseConfiguration(ServerConfiguration conf) {
+    tableIdToTableName = new HashMap<>();
+    poolNameToRegexPattern = new HashMap<>();
+    for (Entry<String,String> table : getTableOperations().tableIdMap().entrySet()) {
```

Comment:
There's logic to handle configuration updates via a Callback. Seems logical that active Balancer could also tie into that callback to process configuration updates. (could be follow-on work since it already doesn't do that)